### PR TITLE
Merge in summaries

### PR DESCRIPTION
**Pull request by mruwnik**
https://github.com/StampyAI/alignment-research-dataset/issues/17

### CODE REVIEW

**Review by ccstan99 — APPROVED** [review]
Very nice! Shouldn't 'authors' and 'summary' just be added to INIT_DICT so they'll be automatically be part of every dataset? We can initialize them to the empty string instead of None, if we don't want to require a non-empty values?

FYI, besides alignment_newsletter, the datasets importai and ml_safety_newsletter are the summarizers for new entries.

File: tests/align_data/common/test_alignment_dataset.py
Comment (Aprillion) [inline tests/align_data/common/test_alignment_dataset.py:1]: (minor) removing tests without replacement? 🙀

File: main.py [hunk 1]
```diff
@@ -41,21 +50,46 @@ def fetch(self, name, rebuild=False) -> None:
...
 
+    def merge_summaries(self, *names):
+        """Update all source materials with summaries if they have any.
+
+        Some datasets are actual alignment content, e.g. arXiv articles, while other datasets are mainly
+        summaries of other articles, e.g. the alignment newsletter. This command merges the two, adding all
+        summaries to all found entries. In theory it's possible for a single article to have multiple different
+        summaries, therefore the summaries are added as a dict of <source>: <summary>
+        """
+        summaries = defaultdict(lambda: dict())
```
[the remark this answers is not in the document]
Comment (mruwnik) [reply main.py:79]: I don't really like dataclasses - I prefer basic dicts whenever possible. Though this it really a Clojure thing which doesn't make all that much sense in python... I can do it with dataclasses if you prefer

File: tests/align_data/common/test_alignment_dataset.py
Comment (mruwnik) [reply tests/align_data/common/test_alignment_dataset.py:1]: it stopped making sense when the distill class was changed - though you're right, of course :D

File: main.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (Aprillion) [reply main.py:79]: I don't have a preference myself, but I have an impression this projects uses dataclasses a lot 😅

File: align_data/arbital/arbital.py [hunk 2]
```diff
@@ -8,6 +8,7 @@
 
 @dataclass
 class Arbital(AlignmentDataset):
+    summary_key: str = 'summary'
```
[the remark this answers is not in the document]
Comment (Aprillion) [reply align_data/arbital/arbital.py:11]: aah 😱 looks like dataclasses are an invasion from Java, now I officially hate them too 😭

File: main.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (mruwnik) [reply main.py:79]: yup... I didn't want to change **too** much, though I suppose I've already overhauled like 70% of the code... :D

**Review by Aprillion — APPROVED** [review]

**Pull request merged**